scripts/gen_dockerfile_test.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python3
+
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit test for gen_dockerfile.py"""
+
+import argparse
+import os
+import re
+import shutil
+import subprocess
+import tempfile
+import unittest
+import unittest.mock
+
+import yaml
+
+import gen_dockerfile
+
+# Expected list of files generated
+EXPECTED_OUTPUT_FILES = ['Dockerfile', '.dockerignore']
+
+
+class GenDockerfileTest(unittest.TestCase):
```

Comment:
Right now, I can run this all on a stock Goobuntu workstation without needed any setup.  That's valuable to me.  Adding a jinja2 or pytest dependency adds friction.  It would be worth it if pytest was 5x better than unittest, but honestly, I think it's about the same or slightly worse (I actually prefer having specific unittest assertions).